lib/checkclass.cpp

```diff
@@ -1551,7 +1565,42 @@ bool CheckClass::hasAllocation(const Function *func, const Scope* scope) const
     return false;
 }
 
-bool CheckClass::hasAssignSelf(const Function *func, const Token *rhs)
+bool CheckClass::isInverseAssignmentTest(const Token *tok)
```

Comment:
hmm.. spontaneously I have the feeling this has some overlap for what we want to achieve with `astIsVariableComparison`. maybe I am wrong.

Reply:
`isInverseAssignmentTest` checks if a self-assignment test is inverse.
For example, for `if (this == &rhs)` and `if (!(this != &rhs))` the function would return `true`, and for 
`if (this != &rhs)` it would return `false`.
Can I achieve this with `astIsVariableComparison`?

Reply:
Done